spec/betagouvbot/webhook_spec.rb

```diff
@@ -0,0 +1,23 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+RSpec.describe BetaGouvBot::Webhook do
+  describe '/compte' do
+    let(:callback)  { 'https://bob.coop' }
+    let(:user_name) { 'bob' }
+    let(:text)      { "#{user_name} #{user_name}@email.coop password" }
+    let(:params)    { { response_url: callback, user_name: user_name, text: text } }
+
+    before do
+      stub_request(:any, callback)
+      stub_request(:any, 'https://beta.gouv.fr/api/v1.3/authors.json')
+        .to_return(
+          headers: { 'content-type' => 'application/json; charset=utf-8' },
+          body: [id: user_name].to_json
+        )
+    end
+
+    it { expect(post('/compte')).not_to be_ok }
+    it { expect(post('/compte', params)).to be_ok }
```

Comment:
That's the idea. But we have to be *very* clear on control flow; Slack behaviour is sensitive to ordering of events and timing. See [API docs](https://api.slack.com/slash-commands).

In particular, we *absolutely* want to send an empty response from the command endpoint, and respond only via the supplied callback URL, when the command parameters include sensitive material; that is the way to suppress in-channel echoing of the original command.

We also want (perhaps) to respond in-channel *before* executing all of the queued actions, so as to beat the 3000ms timeout, otherwise Slack will (temporarily) display an error message to the user, which can be confusing. (Alas, sometimes just waking up the Heroku dyno apparently causes us to miss that window. Not sure yet what to do about that.)